### PR TITLE
Make pinhead client compatible w/ current impl

### PR DESCRIPTION
**Pull request by kahowell**
For convenience of testing, I created `config/application.properties`
(locally, not checked in) with the contents:

```
rhsm-conduit.inventory-service.use-stub=true
rhsm-conduit.pinhead.url=https://${PINHEAD_DEV}/v1
```

replacing `${PINHEAD_DEV}` with the actual dev hostname.

Then, I hit our endpoint with an org, debugged until it worked, working
through a fair number of orgs.

### CODE REVIEW

File: src/main/java/org/candlepin/insights/controller/InventoryController.java [hunk 1]
```diff
@@ -41,65 +42,91 @@
...
-        if (cpuSockets != null) {
+        if (!isEmpty(cpuSockets)) {
             numCpuSockets = Integer.parseInt(cpuSockets);
             facts.setCpuSockets(numCpuSockets);
         }
 
-        String coresPerSocket = consumer.getFacts().get("cpu.core(s)_per_socket");
-        if (coresPerSocket != null && cpuSockets != null) {
+        String coresPerSocket = pinheadFacts.get("cpu.core(s)_per_socket");
+        if (!isEmpty(coresPerSocket) && !isEmpty(cpuSockets)) {
```
Comment (awood) [inline src/main/java/org/candlepin/insights/controller/InventoryController.java]: Would it make sense to combine these a bit?

```java
if (!isEmpty(cpuSockets)) {
    Integer numCpuSockets = Integer.parseInt(cpuSockets);
    facts.setCpuSockets(numCpuSockets);
    if (!isEmpty(coresPerSocket) {
        Integer numCoresPerSocket = Integer.parseInt(coresPerSocket);
        facts.setCpuCores(numCoresPerSocket * numCpuSockets);
    }
}
```

File: src/main/java/org/candlepin/insights/controller/InventoryController.java [hunk 2]
```diff
@@ -41,65 +42,91 @@
...
+        String architecture = pinheadFacts.get("uname.machine");
+        if (!isEmpty(architecture)) {
+            facts.setArchitecture(architecture);
+        }
+    }
+
+    private void extractNetworkFacts(Map<String, String> pinheadFacts, ConduitFacts facts) {
+        String ipAddresses = pinheadFacts.get("Ip-addresses");
+        if (!isEmpty(ipAddresses)) {
+            String[] ipAddressesSplit = ipAddresses.split(", ");
```
Comment (awood) [inline src/main/java/org/candlepin/insights/controller/InventoryController.java]: Let's be a little more robust and split on ",\s*" so the space becomes optional and multiple spaces are eaten by the regex engine

File: src/main/java/org/candlepin/insights/controller/InventoryController.java [hunk 3]
```diff
@@ -41,65 +42,91 @@
...
-            facts.setVirtual(isGuest.equals("True"));
+        String fqdn = pinheadFacts.get("network.fqdn");
+        if (!isEmpty(fqdn)) {
+            facts.setFqdn(fqdn);
         }
 
-        facts.setVmHost(consumer.getHypervisorName());
+        String macAddresses = pinheadFacts.get("Mac-addresses");
+        if (!isEmpty(macAddresses)) {
+            String[] macAddressesSplit = macAddresses.split(", ");
```
Comment (awood) [inline src/main/java/org/candlepin/insights/controller/InventoryController.java]: Same point here about using ",\s*"

File: src/main/java/org/candlepin/insights/controller/InventoryController.java [hunk 4]
```diff
@@ -41,65 +42,91 @@
...
-        List<String> productIds = consumer.getInstalledProducts().stream()
-            .map(installedProduct -> installedProduct.getProductId().toString()).collect(Collectors.toList());
-        facts.setRhProd(productIds);
+    private void extractHypervisorFacts(Consumer consumer, Map<String, String> pinheadFacts,
+        ConduitFacts facts) {
 
-        return facts;
+        String isGuest = pinheadFacts.get("virt.is_guest");
+        if (!isEmpty(isGuest) && !isGuest.equals("Unknown")) {
+            facts.setVirtual(isGuest.equals("True"));
```
Comment (awood) [inline src/main/java/org/candlepin/insights/controller/InventoryController.java]: Worth doing equalsIgnoreCase here?

**Comment by kahowell** [comment]
@awood, rebased and addressed your feedback.

**Pull request merged**